### PR TITLE
Analyzer result improvements

### PR DESCRIPTION
**Pull request by mnonnenmacher**
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/391)

### CODE REVIEW

File: model/src/main/kotlin/AnalyzerResult.kt [hunk 1]
```diff
@@ -74,6 +74,7 @@ data class AnalyzerResult(
             it.dependencies.forEach { addErrors(it) }
         }
 
-        return collectedErrors.filterValues { it.isNotEmpty() }.mapValues { it.value.distinct() }
+        return collectedErrors.filterValues { it.isNotEmpty() }.mapKeys { "${project.id.provider}:${it.key}" }
+                .mapValues { it.value.distinct() }
```
Comment (sschuberth) [inline model/src/main/kotlin/AnalyzerResult.kt]: This starts to get a bit long, and mapping keys / values in two separate runs seems a bit inefficient. Maybe this is better (but I'm not sure myself):

        val mappedErrors = mutableMapOf<String, List<String>>()
        collectedErrors.forEach { key, value ->
            if (value.isNotEmpty()) {
                mappedErrors["${project.id.provider}:$key"] = value.distinct()
            }
        }
        return mappedErrors

File: model/src/main/kotlin/AnalyzerResult.kt [hunk 2]
```diff
@@ -61,7 +61,7 @@ data class AnalyzerResult(
 
     fun collectErrors(): Map<String, List<String>> {
         val collectedErrors = mutableMapOf<String, MutableList<String>>()
-        collectedErrors["Analyzer"] = errors.toMutableList()
+        collectedErrors["${project.id.namespace}:${project.id.name}:${project.id.version}"] = errors.toMutableList()
```
Comment (sschuberth) [inline model/src/main/kotlin/AnalyzerResult.kt]: Why not `project.id.toString()`, which also includes the provider?

File: model/src/main/kotlin/AnalyzerResult.kt (hunk 1 above)
Comment (mnonnenmacher) [reply model/src/main/kotlin/AnalyzerResult.kt]: Done.

File: model/src/main/kotlin/AnalyzerResult.kt (hunk 2 above)
Comment (mnonnenmacher) [reply model/src/main/kotlin/AnalyzerResult.kt]: Because the provider was added below, but I have refactored the whole method now.

**Pull request merged**